src/components/ProductDescription/ProductDescription.js

```diff
@@ -1,20 +1,58 @@
-export default function ProductDescription({ name, price, sku, categoryName, tags, description, specs }) {
-    
-    
-   
+import { useContext, useState } from "react"
+import { CartItemsContext } from '../CartItemsContextProvider/CartItemsContextProvider'
+
+export default function ProductDescription({item, name, price, sku, categoryName, tags, description, specs, stock, uniqueId }) {
+
+
+    const { cartItems, setCartItems } = useContext(CartItemsContext);
+    const [numberOfItems, setNumberOfItems] = useState(1);
+
+    function addItemToCard() {
+        let totalElements = parseInt(numberOfItems, 10);
+        let elements = cartItems;
+        let cartItem = cartItems.find((item) => item.uniqueId === uniqueId);
+        if (cartItem !== undefined) {
+            totalElements += cartItem.quantity;
+            elements = cartItems.filter((item) => item.uniqueId !== uniqueId);
+        }
+        else {
+            cartItem = {
+                uniqueId: uniqueId,
+                quantity: 0,
+                item:{...item,name, price, sku, categoryName, tags, description, specs, stock, uniqueId }
+            }
+        }
+        if (totalElements > stock) {
+            setTimeout(() => {
+                alert('not enough stock');
```

Comment:
This is not correct, you should disable the add button instead of display an alert.